### PR TITLE
Enhance AI Feature Management and UI/UX Improvements

### PR DESCRIPTION
**Pull request by ndoschek**
#### What it does

This PR introduces several enhancements and fixes related to the AI features in the project. The key changes include:

- AI Feature Management: Implemented a master setting to enable/disable view contributions, consolidated all AI settings under a single category, and introduced a service for AI activation.
- User Interface Updates: Added a banner on the welcome page and a simple message in the AI chat view for disabled features. Marked all AI-related views as "Experimental" and addressed minor UI/UX issues.
- Bug Fixes: Resolved issues with AI code completion preferences and the toggle functionality for AI completion.

These updates aim to streamline the user experience and provide clearer information about the experimental status of AI features.

Resolves https://github.com/eclipsesource/osweek-2024/issues/71

Co-authored-by: Tobias Ortmayr <tortmayr@eclipsesource.com>

#### Follow-ups

https://github.com/eclipsesource/osweek-2024/issues/110

#### Review checklist

- [x] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

File: packages/ai-chat-ui/src/browser/chat-view-widget.tsx [hunk 1]
```diff
@@ -71,14 +79,25 @@ export class ChatViewWidget extends BaseWidget implements StatefulWidget {
...
         ]);
         const layout = this.layout = new PanelLayout();
+
+        // Experimental features are enabled
         this.treeWidget.node.classList.add('chat-tree-view-widget');
         layout.addWidget(this.treeWidget);
         this.inputWidget.node.classList.add('chat-input-widget');
         layout.addWidget(this.inputWidget);
         this.inputWidget.onQuery = this.onQuery.bind(this);
+        this.inputWidget.setEnabled(false);
```
Comment (CamilleLetavernier) [inline packages/ai-chat-ui/src/browser/chat-view-widget.tsx:89]: We should check the preference here. Having a listener might cause the feature to be enabled, but the view remains disabled (until you toggle the preference again).

Not sure how to reproduce this exactly, but it happened to me this morning when coming back to the page after laptop went to sleep: the setting was enabled, but the chat was disabled.

File: packages/ai-core/src/browser/index.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+// terms of the Eclipse Public License v. 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0.
+//
+// This Source Code may also be made available under the following Secondary
+// Licenses when the conditions for such availability set forth in the Eclipse
+// Public License v. 2.0 are satisfied: GNU General Public License, version 2
+// with the GNU Classpath Exception which is available at
+// https://www.gnu.org/software/classpath/license.html.
+//
+// SPDX-License-Identifier: EPL-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0
```
Comment (CamilleLetavernier) [inline packages/ai-core/src/browser/index.ts:14]: Last line of the copyright header is missing. We need more stars in this PR!

File: packages/getting-started/src/browser/getting-started-widget.tsx [hunk 3]
```diff
@@ -131,6 +133,9 @@ export class GettingStartedWidget extends ReactWidget {
     protected render(): React.ReactNode {
         return <div className='gs-container'>
             <div className='gs-content-container'>
+                <div className='gs-float shadow-pulse'>
```
Comment (CamilleLetavernier) [inline packages/getting-started/src/browser/getting-started-widget.tsx:136]: Should we have some kind of check to make sure the AI Packages are actually installed, before rendering this section?

**Review by CamilleLetavernier — COMMENTED** [review]
Looks great ✨! I only have some minor comments (inline)

**Review by sdirix — APPROVED** [review]

**Pull request merged**